server/src/modules/admin/admin.service.ts

```diff
@@ -185,4 +187,27 @@ export class AdminService {
 
         await this.userRepo.delete(userID);
     }
+
+    async getReports(skip?: number, take?: number, resolved?: boolean) {
+        const where = { resolved: resolved };
+        const dbResponse = await this.userRepo.getAllReports(where, skip, take);
+        return new PaginatedResponseDto(ReportDto, dbResponse);
+    }
+
+    async updateReport(userID: number, reportID: number, reportDto: UpdateReportDto) {
+        const reports = await this.userRepo.getAllReports({ id: reportID });
+
+        if (reports[1] === 0) throw new NotFoundException('Report not found');
+
+        const where: Prisma.ReportWhereUniqueInput = {
+            id: reportID
+        };
+        const data: Prisma.ReportUpdateInput = {
+            resolved: reportDto.resolved,
+            resolutionMessage: reportDto.resolutionMessage
+        };
+        if (reportDto.resolved) data.resolver = { connect: { id: userID } };
+
+        return await this.userRepo.updateReports(where, data);
```

Comment:
This endpoint 204s (iirc it's compliance with old API, PATCHs don't respond with content), so leave out the `return` here, just do `await this.userRepo.updateReports(where, data);` so it trivially returns `Promise<void>`